### ISSUE TITLE
[Core, Spell, Mechanic] Cleaves - Range is too small

### ISSUE THREAD

**Issue by Aridam** [SunwellTracker/issues#2670]
Decription: 
An issue with Warrior's Cleave, resulting in a too small Cleave Range.

How it works: 
Cleave Range is too small.
https://gfycat.com/enchantedflamboyantbasilisk

How it should work: 
It's supposed to be much bigger.

Source (you should point out proofs of your report, please give us some source):
https://www.youtube.com/watch?v=3z2nV3pRDS0&t=3m40s (TBC)
https://www.youtube.com/watch?v=8WnaIKa6weY&feature=youtu.be&t=1m59s
https://www.youtube.com/watch?v=1mYHEzHuld8&feature=youtu.be&t=39
https://www.youtube.com/watch?v=s5KLetVQn38&feature=youtu.be&t=1m17s
https://www.youtube.com/watch?v=u4qBKWWbRC4&feature=youtu.be&t=120
https://www.youtube.com/watch?v=DmwUI5-lDWo&feature=youtu.be
https://www.youtube.com/watch?v=oYhPEjWZDik&feature=youtu.be&t=50s (Cata pre-patch)

Steps to reproduce:
1. Have 2+ Targets at the ready;
2. Use Cleave;
3. Repeat with wider and wider gaps between the Targets.

**Easiest way to reproduce would be to copy [this](https://www.youtube.com/watch?v=DmwUI5-lDWo&feature=youtu.be) -- which is what I did under "How it works".
Additionally, you can test as done in any of the other videos -- you should get no Secondary Target Cleave hits as done in any of them, 'cause the Range is too short here. Not sure if it's because perhaps Cleave hits from the center of Hitboxes instead of their edge, or some sort of wrong cone width, but something's _definitely_ wrong here.**

Additional information:
<details>
<summary>Cleave Spell IDs:</summary>

https://db.darkwizard.pl/?spell=845
https://db.darkwizard.pl/?spell=7369
https://db.darkwizard.pl/?spell=11608
https://db.darkwizard.pl/?spell=11609
https://db.darkwizard.pl/?spell=20569
https://db.darkwizard.pl/?spell=25231
https://db.darkwizard.pl/?spell=47519
https://db.darkwizard.pl/?spell=47520
</details>

**Comment by Riztazz** [comment]
Interesting, i wonder which additional sizes come in play when blizzard calculates those

**Comment by styxel** [comment]
I have had the same feeling with swipe in cat form but im not 100% sure.

**Comment by Aridam** [comment]
> Interesting, i wonder which additional sizes come in play when blizzard calculates those

I believe this -- as with other AoE (such as PBAoE) -- depends on sizes and something regarded to as "Combat Reach" _or_ "Melee Reach". For example as you can see in the Halion timestamps, it's a big Dragon Boss with a huge Hitbox, and you can additionally hit from so far past that Hitbox that you can make Twilight Cutters seem a joke to Melee, and you can see how Cleaves work on him and off him to surrounding Living Blazes (observe any Whirlwinds happening too, for reference -- that's a PBAoE). The Anub'arak example is similar.

This may or may not be useful, but I used this info as referenced [here](https://github.com/SunwellTracker/issues/issues/2544) using the help of a TC report I linked there -- it's possible this may be applied to other AOEs like Cleave as well, but who knows -- if it helps pinpoint this, then good. It's only _clear_ that things are wrong.

> I have had the same feeling with swipe in cat form but im not 100% sure. Like on anub 25man i cant swipe the 2 adds +boss unless they areclike really on top of each other or In heroic Dungeons u have To reposition urself all the time because your swipe only hits 1 target for some reason and this results In shit aoe dps often because ur wasting all that energy for 1 hit

Hmmm, so maybe this affects Cleaves as a whole, rather than being merely Warrior-related. The fact you mentioned that they need to be on top of each-other -- it seems to support my theory that it's taking it from the center of Hitboxes rather than the edges, and that's considering those Hitboxes can be huge depending on what you're fighting. So, as a result, fighting huge Bosses can actually be a hindrance for Cleaves.

Seeing as how far those Cleaves can go may also mean that indeed there's Caster/Target sizes and Combat/Melee Reach of both accounted for, otherwise all those big Targets would impose issues and force people to re-Target to avoid messing up.

**Comment by styxel** [comment]
https://youtu.be/3YxJWuQ6wHo at 0.25 In the video i bet this isnt possible on Frosthold but im not sure cant test stuff now. But yea something If definately wrong.

**Comment by Aridam** [comment]
[This](https://www.youtube.com/watch?v=8WnaIKa6weY&feature=youtu.be&t=1m59s) and [this]https://www.youtube.com/watch?v=oYhPEjWZDik&feature=youtu.be&t=50s) (despite being v4.0.1) indicate it _should_ be possible. Cleave Range seems to have been consistent for a long time. They (and some of the other provided examples) also show that it is possible for Cleave to hit Secondary Targets which are out of _your_ Melee Range.

**Comment by styxel** [comment]
https://gyazo.com/78e09fd4c7fcf784933c08cfcef70330 able to hit everything with melee but swipe only hits the mid one

**Comment by Aridam** [comment]
Good showcase there. That should've worked, like in the original video showcasing Swipe on Theramore Dummies.

Also see this:
https://www.youtube.com/watch?v=HnUSYSjoRTk&t=11m40s
The Secondary Targets are way-off from the Warrior's own Melee Range, but get hit by Cleave with the Main Targets as reference. The way it looks from this PoV is it seems to take out from the edge of something's Hitbox, see if something's within it's Combat/Melee Range using _its_ Hitbox edge, resulting this massive extension. At least, that's how it sounds if put logically.

It seems Cleaves' jumps are Target-based -- they can bounce from one to the other and exceed your own Melee Range to any and all Secondary Targets, so long as the STs are close to the MTs.

**Comment by blackhodown** [comment]
This is the same for Heart Strike on Blood DKs. Hitbox is absurdly tiny, much smaller than it is supposed to be.

**Comment by Aridam** [comment]
So it seems to affect Cleaves in general. Gonna change title.

**Comment by Aridam** [comment]
@styxel 
Provide some quick Bear Swipe and Maul test results to add to this if you can.

I'll go respec to Prot to test Improved Revenge when I can.

**Comment by styxel** [comment]
i will later swipe bear has higher range than cat though being 8 yards when swipe cat is 5

**Comment by styxel** [comment]
https://gyazo.com/c1458e1a19cfa6790dde9adc340f3afb maul likes to hit the one in the back and swipe the target next to it

**Comment by Aridam** [comment]
Well that's weird.

I went back to those Ironforge Dummies and kept trying to align Cleaves on them -- worked only once in a very-finicky spot and angle, which I then couldn't reproduce anymore because I don't even know how I pulled it off. I think -- for Cleaves with facing/angle requirements -- the width of the angle where you're facing is screwed here.

**Comment by Aridam** [comment]
Update: There definitely seems to be an issue with the angle.

See this:
https://gfycat.com/infiniteradiantbunny

The angle is definitely narrower than intended -- compare with the provided evidence as to how it worked on Retail. It looks to be 180 degrees (entire frontal arc), while here it's evidently less than that.

**Comment by Riztazz** [comment]
Those adjectives are unnecessary. I see the code related to those spells and i'm pretty sure it fixes it but gotta test a few cases first, if everything is fine then it will be live on monday

**Comment by Riztazz** [comment]
Still got some issues i have to take care of, postponed till tomorrow.
This one is weird..

**Comment by Riztazz** [comment]
That fury warrior DPSing in orgrimmar, any chance we could get a similar video where he cleaves but warrior is facing the other dummy (on the left)? That would confirm a lot of things.

Dummy has a combat reach of 2.5y, so now the main question is, how big is cleave radius excluding combat reach and what if creature has gigantic combat reach, some has them far exceeding their bounding box (circle around creature)

**Comment by Aridam** [comment]
> That fury warrior DPSing in orgrimmar, any chance we could get a similar video where he cleaves but warrior is facing the other dummy (on the left)? That would confirm a lot of things.

Will try, but that alone was difficult to find -- something as specific as that might be even moreso.

> Dummy has a combat reach of 2.5y, so now the main question is, how big is cleave radius excluding combat reach and what if creature has gigantic combat reach, some has them far exceeding their bounding box (circle around creature)

The provided Halion videos show this interaction -- you can watch a few seconds before/after the posted timestamps to see Landsoul repositioning and _still_ being able to Cleave and have it reach around. See Halion's Hitbox, yet see how far Landsoul is able to hit from and land proper Cleaves with changes in position and facing angle.

> TC has base cleave radius of 5y, we have base of 8y, corecraft had that at 10y as base.
So assuming combat reach is added then 5y is wrong, dummies would never get hit in this case.
8y and 10y covers both dummies (left and right) depending on facing.

TC's is definitely wrong, exactly for the reason you pointed out, especially given the videos for comparison. Now we need to see if it's 8yd or 10yd.

This:
https://www.youtube.com/watch?v=8WnaIKa6weY&feature=youtu.be&t=1m59s
might be able to provide some more insight if I go to the same spot and check how far that other Dummy would be from me.

> And what if creature has combat reach of 30yards, just because, should it hit a creature 38y from center of the target? My guess is that combat reach can't exceed base radius of a spell, so additional yards would get cut to 8y if reach exceeds 8

Is there a Mob with a ridiculously-long Combat Reach than Halion's/Sindragosa's, perhaps? I could try to find some footage with it if there is and see how Cleaves interact with it.

**Comment by Riztazz** [comment]
There is a lot of creatures that are small but have gigantic combat reaches for whatever reason :P

**Comment by Aridam** [comment]
Erm, not sure how that list really helps me -- would rather have an NPC ID list with high Combat Reach, so I can find _what_ they actually are and then trace each and every one like that via footage with possible Cleaves.

**Comment by Riztazz** [comment]
Filter it out via wowhead? :P

List of creatures that have combat reach either 8 or higher than 8.
DisplayId || Combat reach
```
(800, 9);
(2718, 10);
(2719, 10);
(2837, 10);
(4872, 10);
(7271, 15);
(8570, 14);
(9832, 10);
(10193, 12);
(10244, 20);
(11121, 20);
(11380, 17);
(11460, 17);
(12869, 10);
(13992, 13);
(14173, 12);
(14367, 9);
(14992, 14);
(15288, 8.5);
(15295, 10.5);
(15345, 12);
(15363, 18);
(15364, 18);
(15365, 18);
(15366, 18);
(15392, 9);
(15432, 12.5);
(15473, 17.5);
(15509, 18.75);
(15556, 15);
(15686, 10);
(15786, 25);
(15928, 20.01);
(15931, 10);
(15945, 10);
(15948, 10);
(16033, 22);
(16064, 12);
(16137, 10);
(16630, 15);
(17180, 8.004);
(17205, 10);
(17228, 10.5);
(17345, 10);
(17445, 12.5);
(18062, 10.5);
(18155, 17.5);
(18194, 9);
(18224, 10.5);
(18526, 8.8);
(18527, 12);
(18622, 13.2);
(18671, 20);
(18698, 10);
(18839, 13.5);
(18945, 12);
(18951, 9);
(19162, 15);
(19270, 27.5);
(19501, 12.5);
(19537, 10);
(19581, 10);
(19585, 14);
(19667, 10);
(19677, 10);
(19691, 10);
(19716, 20);
(19797, 10);
(19943, 9);
(20127, 27);
(20216, 22);
(20346, 10);
(20477, 20);
(20522, 14);
(20592, 11.9);
(20593, 10);
(20645, 12);
(20770, 9);
(20916, 9);
(20975, 9);
(21018, 10);
(21145, 18);
(21174, 10.5);
(21213, 8.75);
(21426, 10);
(21435, 15);
(22355, 22);
(22711, 18);
(22838, 10);
(22965, 17.5);
(23200, 12);
(23344, 18);
(23345, 10.5);
(24464, 15);
(27736, 9);
(27897, 10.5);
(21601, 12);
(21603, 30);
(22721, 10);
(23214, 10.5);
(23355, 8.75);
(23414, 10);
(23704, 25);
(23784, 12);
(23881, 10);
(23883, 10);
(23895, 10);
(23964, 9);
(23970, 77);
(24137, 10);
(24140, 10);
(24175, 12.5);
(24176, 15);
(24190, 10);
(24272, 16.5);
(24301, 9);
(24417, 15);
(24782, 9);
(24808, 35);
(24849, 9);
(24850, 9);
(24854, 9);
(24873, 12);
(24876, 15);
(24892, 9);
(24934, 35);
(24954, 12);
(25096, 12);
(25129, 12.5);
(25194, 14);
(25215, 9);
(25254, 10);
(25400, 15);
(25515, 15);
(25634, 10.5);
(25685, 10.5);
(26100, 10);
(26270, 30);
(26284, 10);
(26491, 8.75);
(26534, 22.5);
(26536, 20);
(26656, 12);
(26699, 15);
(26752, 20);
(26778, 9);
(26967, 9);
(27034, 20);
(27035, 10);
(27093, 50);
(27108, 9);
(27176, 40);
(27225, 10);
(27271, 8.75);
(27280, 14);
(27338, 20);
(27340, 10);
(27419, 9);
(27570, 20);
(27811, 10);
(27813, 10);
(27856, 10);
(27951, 12);
(27979, 18);
(27982, 9);
(27992, 10);
(28008, 10);
(28013, 12);
(28517, 10);
(28611, 10);
(28638, 22);
(28641, 9);
(28787, 15);
(28817, 20);
(28821, 25);
(28822, 25);
(28875, 15);
(28883, 9);
(28887, 22.5);
(28951, 30);
(29096, 8.5);
(29104, 20);
(29185, 12);
(29878, 9);
(29267, 9);
(29240, 9);
(29487, 33);
(29524, 9);
(29268, 10);
(30318, 20);
(30357, 8.1);
(30362, 18.75);
(30483, 9);
(30521, 12);
(30652, 19.5);
(30902, 15);
(31089, 9);
(31119, 20.25);
(32179, 12.25);
(31761, 9);
(31952, 18);
```


DisplayId | Entry | Name
```
(2837, 4829, 'Aku\'mai');
(11460, 6109, 'Azuregos');
(4872, 6140, 'Hetaera');
(7271, 7273, 'Gahz\'rilla');
(8570, 10184, 'Onyxia');
(9832, 10415, 'Ash\'ari Crystal');
(10193, 10430, 'The Beast');
(10244, 10942, 'Nessy');
(14173, 11496, 'Immol\'thar');
(11121, 11502, 'Ragnaros');
(11380, 11583, 'Nefarian');
(10193, 11982, 'Magmadar');
(13992, 13020, 'Vaelastrasz the Corrupt');
(14367, 14020, 'Chromaggus');
(14992, 14435, 'Prince Thunderaan');
(15295, 14834, 'Hakkar');
(15364, 14887, 'Ysondre');
(15365, 14888, 'Lethon');
(15366, 14889, 'Emeriss');
(15363, 14890, 'Taerar');
(15288, 15114, 'Gahz\'ranka');
(15288, 15122, 'Gahz\'ranka Dead');
(2719, 15192, 'Anachronos');
(15345, 15263, 'The Prophet Skeram');
(15686, 15299, 'Viscidus');
(800, 15305, 'Lord Skwol');
(15948, 15334, 'Giant Eye Tentacle');
(15432, 15339, 'Ossirian the Unscarred');
(15392, 15340, 'Moam');
(15509, 15517, 'Ouro');
(15556, 15589, 'Eye of C\'Thun');
(15363, 15689, 'Netherspite');
(16137, 15928, 'Thaddius');
(16064, 15932, 'Gluth');
(15928, 15952, 'Maexxna');
(15931, 15956, 'Anub\'Rekhan');
(16033, 15989, 'Sapphiron');
(15945, 15990, 'Kel\'Thuzad');
(19581, 17133, 'Aged Clefthoof');
(18062, 17225, 'Nightbane');
(18527, 17257, 'Magtheridon');
(17180, 17683, 'Zarakh');
(21435, 17711, 'Doomwalker');
(17228, 17770, 'Hungarfen');
(18526, 17842, 'Azgalor');
(18194, 17882, 'The Black Stalker');
(18224, 17942, 'Quagmirran');
(19585, 18259, 'Banthar');
(19797, 18398, 'Brokentoe');
(18155, 18411, 'Durn the Hungerer');
(19537, 18544, 'Veraku');
(17445, 18678, 'Fulgorge');
(20770, 18680, 'Marticar');
(18839, 18708, 'Murmur');
(18671, 18733, 'Fel Reaver');
(18622, 18945, 'Pit Commander');
(18698, 19044, 'Gruul the Dragonkiller');
(19162, 19219, 'Mechano-Lord Capacitus');
(16630, 19400, 'Fel Reaver Sentry');
(18945, 19514, 'Al\'ar');
(18951, 19516, 'Void Reaver');
(19270, 19934, 'Zaladormu');
(19667, 20216, 'Grulloc');
(18671, 20243, 'Scrapped Fel Reaver');
(19537, 20332, 'Nether Dragon');
(19677, 20555, 'Goc');
(19691, 20600, 'Maggoc');
(19716, 20682, 'Terokkarantula');
(9832, 20899, 'Void Conduit');
(19943, 20912, 'Harbinger Skyriss');
(20522, 21102, 'Uvuros');
(20127, 21174, 'Magtheridon');
(20216, 21217, 'The Lurker Below');
(19581, 21313, 'Dead Clefthoof');
(19501, 21404, 'Legion Hold Fel Reaver');
(19943, 21466, 'Harbinger Skyriss');
(19943, 21467, 'Harbinger Skyriss');
(20346, 21514, 'Gorgrom the Dragon-Eater');
(20593, 21657, 'Neltharaku');
(20477, 21801, 'Vhel\'kur');
(20592, 21897, 'Felspine the Greater');
(20645, 21961, 'Cataclysm Overseer');
(15786, 22137, 'Summoned Old God');
(18194, 22174, 'Trigul');
(18671, 22196, 'Wrath Reaver');
(20346, 22199, 'Slaag');
(18526, 22281, 'Galvanoth');
(20975, 22283, 'Eredar Stormbringer');
(19797, 22287, 'Amberpelt Clefthoof');
(18671, 22293, 'Inactive Fel Reaver');
(18671, 22295, 'Deathforge Automaton');
(20916, 22297, 'Throne-Guard Highlord');
(18526, 22302, 'Throne-Guard Champion');
(18671, 22389, 'Kil\'Jaeden Reaver (non-interactable)');
(21018, 22441, 'Teribus the Cursed');
(21174, 22887, 'High Warlord Naj\'entus');
(21145, 22898, 'Supremus');
(21213, 23029, 'Talonsworn Forest-Rager');
(15473, 15481, 'Spirit of Azuregos');
(15786, 15809, 'C\'Thun Transformation Visual');
(17445, 17075, 'Sandworm');
(16630, 18060, 'Fel Reaver Netherstorm');
(16630, 19260, 'Destroyed Fel Reaver');
(19943, 21601, 'Harbinger Skyriss (1)');
(19943, 21600, 'Harbinger Skyriss (1)');
(19943, 21599, 'Harbinger Skyriss (1)');
(19162, 21533, 'Mechano-Lord Capacitus (1)');
(18839, 20657, 'Murmur (1)');
(18671, 21187, 'Designer Island Fel Reaver [PH]');
(19537, 21407, 'Netherwing Dragon');
(18194, 20184, 'The Black Stalker (1)');
(18194, 22056, 'Coilfang Strider');
(17228, 20169, 'Hungarfen (1)');
(19501, 22509, 'Scrapped Fel Reaver Transform');
(19677, 22910, 'Skulloc Soulgrinder');
(20975, 23230, 'Shartuul');
(12869, 23261, 'Furywing');
(12869, 23281, 'Insidion');
(12869, 23282, 'Obsidia');
(12869, 11978, '[NOT USED] Neltharion');
(11121, 12804, '[PH] TEST Fire God');
(19581, 22284, '[PH] Wrath Clefthoof [not used]');
(14173, 22930, 'Yor');
(18224, 19894, 'Quagmirran (1)');
(21426, 23348, 'Captain Skyshatter');
(22355, 23899, 'Tethyr');
(22711, 25158, 'Brutallus');
(22965, 25160, 'Madrigosa');
(23345, 24850, 'Kalecgos');
(22711, 24882, 'Brutallus');
(22965, 24895, 'Madrigosa');
(22838, 25038, 'Felmyst');
(22838, 25041, 'Felmyst Visual');
(23200, 25315, 'Kil\'jaeden');
(23345, 25319, 'Kalecgos');
(24140, 25500, 'PattyMacks Flying Dummy');
(17205, 25508, 'Shadowsword Guardian');
(14173, 25599, 'Cataclysm Hound');
(17205, 25954, 'Shadowsword Guardian Sunwell');
(14173, 25957, 'Cataclysm Hound Sunwell');
(21601, 23517, 'Northrend Yeti (White)');
(21603, 23519, 'Big Shirl');
(24137, 23538, 'Northrend Red Dragon');
(16033, 24019, 'Glacion');
(24140, 24770, 'Nexus Watcher');
(26699, 24795, 'Surristrasz');
(19797, 24894, 'Bluff Behemoth');
(22721, 24899, '"Scoodles"');
(23355, 24914, 'Sorlof');
(25194, 25203, 'Glrggl');
(23883, 25228, 'Risen Crypt Lord');
(24873, 25430, 'Magmothregar');
(23214, 25465, 'Kel\'Thuzad');
(24849, 25789, 'Gammothra the Tormentor');
(24849, 25790, 'Weakened Gammothra');
(23414, 25833, 'The Grinder');
(27280, 25862, 'Khu\'nok the Behemoth');
(23704, 26231, 'Saragosa');
(15509, 26250, 'Scourged Burrower');
(27271, 26271, 'Emaciated Mammoth Bull');
(24140, 26276, 'Nexus Guardian');
(27811, 26277, 'Bronze Shrine Warden');
(24137, 26279, 'Dragonblight Red Dragon');
(23784, 26290, 'Jotun');
(27176, 26310, 'Malygos');
(24876, 26360, 'Rattlebore');
(22711, 26376, 'Test - Brutallus Craig');
(20216, 26452, 'Leviroth');
(23895, 26521, 'Kili\'ua');
(23881, 26608, 'Under-King Anub\'et\'kan');
(23964, 26633, 'Ursoc');
(23970, 26648, 'Oacha\'noa');
(26284, 26686, 'Ferocious Rhino');
(27419, 26687, 'Gortok Palehoof');
(24954, 26836, 'Gigantaur');
(24190, 26949, 'Torastrasza');
(24854, 27002, 'Grom\'thar the Thunderbringer');
(24850, 27004, 'Icefist');
(24850, 27005, 'Chilltusk');
(24854, 27006, 'Bonesunder');
(24850, 27007, 'Iceshatter');
(24854, 27008, 'Bloodfeast');
(24854, 27009, 'Drakegore');
(24272, 27075, 'Dun Argol Power Core');
(24301, 27105, 'Kreug Oathbreaker');
(24464, 27199, 'Warlord Jin\'arrak');
(24417, 27212, 'Image of Loken');
(23964, 27372, 'Ursoc Credit');
(24137, 27530, 'Ruby Keeper');
(24140, 27608, 'Azure Dragon');
(27034, 27656, 'Ley-Guardian Eregos');
(24808, 27789, 'Ysera');
(23964, 27796, 'Worg\'s Blood Elixir Credit');
(24934, 27925, 'Nozdormu');
(2837, 28003, 'Bittertide Hydra');
(25634, 28030, 'Quetz\'lun\'s Spirit');
(25096, 28068, 'Prophet of Sseratus');
(25129, 28089, 'Sseratus');
(25215, 28212, 'Bythius the Flesh-Shaper');
(25254, 28243, 'Thrym');
(28008, 28401, 'Har\'koa');
(25400, 28416, 'Rhunok');
(25515, 28561, 'Spirit of Rhunok');
(15295, 28672, 'Quetz\'lun');
(25685, 28785, 'Spirit of Quetz\'lun');
(26752, 28859, 'Malygos');
(27035, 28860, 'Sartharion');
(26270, 28952, 'Akali');
(27856, 29120, 'Anub\'arak');
(15931, 29249, 'Anub\'Rekhan (1)');
(15928, 29278, 'Maexxna (1)');
(26100, 29280, 'Image of Arthas');
(10193, 29312, 'Lavanthor');
(27340, 29317, 'Cyanigosa (Dragon)');
(24892, 29390, 'Snowdrift Jormungar');
(16064, 29417, 'Gluth (1)');
(16137, 29448, 'Thaddius (1)');
(28013, 29503, 'Fjorn');
(26656, 29647, 'Gymer');
(26536, 29747, 'The Ocular');
(26491, 29748, 'Phantom Mammoth');
(26534, 29789, 'The Ocular - Eye of C\'Thun Transform');
(26656, 29884, 'Gymer');
(25254, 29895, 'Thrym');
(16033, 29991, 'Sapphiron (1)');
(24892, 30014, 'Yggdras');
(24954, 30017, 'Stinkbeard');
(15945, 30061, 'Kel\'Thuzad (1)');
(21603, 30080, 'Willzyx');
(19797, 30088, 'Brokentoe (Mount)');
(26778, 30105, 'King Jokkum');
(30652, 30294, 'Frostworg Denmother');
(27093, 30300, 'Iron Colossus');
(26778, 30331, 'King Jokkum');
(26967, 30341, 'Eisenfaust');
(27225, 30453, 'Onyx Sanctum Guardian');
(23883, 30541, 'Forgotten Depths Underking');
(23883, 30544, 'Forgotten Depths Underking');
(26284, 30770, 'Ferocious Rhino (1)');
(27419, 30774, 'Gortok Palehoof (1)');
(26100, 30778, 'Image of Arthas (1)');
(23970, 30794, 'Oacha\'noa (1)');
(27338, 30830, 'Underking Talonox');
(27225, 31001, 'Onyx Sanctum Guardian (1)');
(23883, 31039, 'Forgotten Depths Underking');
(26967, 31125, 'Archavon the Stone Watcher');
(24301, 31255, 'Saronite Shaper');
(24301, 31271, 'Carnage');
(27035, 31311, 'Sartharion (1)');
(24301, 31322, 'Saronite Shaper');
(27570, 31334, 'Korialstrasz');
(24176, 31393, 'Crystal Wyrm');
(10193, 31509, 'Lavanthor (1)');
(27034, 31561, 'Ley-Guardian Eregos (1)');
(27856, 31610, 'Anub\'arak (1)');
(17345, 31688, 'Whirlwind');
(26967, 31722, 'Archavon the Stone Watcher (1)');
(26752, 31734, 'Malygos');
(17345, 31765, 'Tidal Wave');
(23883, 31780, 'Fallen Spiderlord');
(27736, 31815, 'Bone Giant');
(27813, 32186, 'Infinite Timebreaker');
(10193, 32237, 'Lava Hound');
(27897, 32299, 'Bone Sentinel');
(27951, 32400, 'Tukemuth');
(27897, 32479, 'Bone Guard');
(23883, 32480, 'Nerubian Underking');
(24301, 32482, 'Pustulent Colossus');
(27992, 32483, 'Engorged Blight Worm');
(27979, 32487, 'Putridus the Ancient');
(27982, 32492, 'Frostbrood Matriarch');
(26778, 32540, 'Lillehoff');
(10193, 32553, 'Lava Hound (1)');
(28641, 32871, 'Algalon the Observer');
(28638, 32930, 'Kologarn');
(28821, 32933, 'Left Arm');
(28822, 32934, 'Right Arm');
(28641, 33070, 'Algalon the Observer (1)');
(29096, 33090, 'Pool of Tar');
(28875, 33113, 'Flame Leviathan');
(29185, 33118, 'Ignis the Furnace Master');
(28883, 33139, 'Flame Leviathan Turret');
(28787, 33186, 'Razorscale');
(29185, 33190, 'Ignis the Furnace Master (1)');
(28517, 33237, 'Ulduar Colossus');
(28817, 33288, 'Yogg-Saron');
(28611, 33293, 'XT-002 Deconstructor');
(28887, 33329, 'Heart of the Deconstructor');
(27982, 33687, 'Chillmaw');
(2718, 33716, 'Ruby Consort');
(2719, 33718, 'Bronze Consort');
(12869, 33720, 'Obsidian Consort');
(28787, 33724, 'Razorscale (1)');
(28611, 33885, 'XT-002 Deconstructor (1)');
(28951, 33890, 'Brain of Yogg-Saron');
(28638, 33909, 'Kologarn (1)');
(28821, 33910, 'Left Arm (1)');
(28822, 33911, 'Right Arm (1)');
(28951, 33954, 'Brain of Yogg-Saron (1)');
(28817, 33955, 'Yogg-Saron (1)');
(27108, 33993, 'Emalon the Storm Watcher');
(27108, 33994, 'Emalon the Storm Watcher (1)');
(28887, 33995, 'Heart of the Deconstructor (1)');
(28875, 34003, 'Flame Leviathan (1)');
(28517, 34105, 'Ulduar Colossus (1)');
(28883, 34111, 'Flame Leviathan Turret (1)');
(26778, 34133, 'Champion of Hodir');
(26778, 34139, 'Champion of Hodir (1)');
(29104, 34164, 'Mechagnome Battletank');
(29104, 34165, 'Mechagnome Battletank (1)');
(28638, 34297, 'Kologarn');
(28638, 34298, 'Kologarn (1)');
(21601, 35470, 'Icehowl');
(24892, 35482, 'Hungry Jormungar');
(18698, 35039, 'Memory of Gruul');
(29267, 36066, 'Eydis Darkbane');
(29240, 36065, 'Fjola Lightbane');
(29487, 34925, 'North Sea Kraken');
(21601, 34797, 'Icehowl');
(29240, 34497, 'Fjola Lightbane');
(29267, 34496, 'Eydis Darkbane');
(29524, 35013, 'Koralon the Flame Watcher');
(29268, 34564, 'Anub\'arak');
(31119, 36612, 'Lord Marrowgar');
(27982, 36661, 'Rimefang');
(27982, 36794, 'Scourgelord Tyrannus');
(30902, 37230, 'Spire Frostwyrm');
(27982, 37528, 'Spire Frostwyrm (Ambient)');
(30362, 37755, 'Sindragosa');
(30483, 37025, 'Stinky');
(30483, 37217, 'Precious');
(14367, 35033, 'Memory of Chromaggus');
(30318, 36789, 'Valithria Dreamwalker');
(27982, 37533, 'Rimefang');
(27982, 37534, 'Spinestalker');
(30318, 37950, 'Valithria Dreamwalker');
(31089, 38433, 'Toravon the Ice Watcher');
(30521, 37126, 'Sister Svalna');
(8570, 36538, 'Onyxia (1)');
(18698, 35527, 'Memory of Gruul (1)');
(14367, 35521, 'Memory of Chromaggus (1)');
(27982, 38220, 'Rimefang (1)');
(30362, 38267, 'Sindragosa (3)');
(30902, 38444, 'Spire Frostwyrm (1)');
(30483, 38064, 'Stinky (1)');
(30483, 38103, 'Precious (1)');
(31119, 37959, 'Lord Marrowgar (3)');
(30318, 38174, 'Valithria Dreamwalker (1)');
(27982, 38219, 'Spinestalker (1)');
(30521, 38258, 'Sister Svalna (1)');
(28611, 34515, 'XT-005 Debugger');
(30362, 38266, 'Sindragosa (2)');
(31119, 37958, 'Lord Marrowgar (2)');
(30362, 38265, 'Sindragosa (1)');
(28517, 36168, 'Dan\'s Test Colossus');
(31119, 37957, 'Lord Marrowgar (1)');
(21601, 35447, 'Icehowl (1)');
(29240, 35350, 'Fjola Lightbane (1)');
(29267, 35347, 'Eydis Darkbane (1)');
(29268, 34566, 'Anub\'arak (1)');
(21601, 35448, 'Icehowl (2)');
(29240, 35351, 'Fjola Lightbane (2)');
(29267, 35348, 'Eydis Darkbane (2)');
(29268, 35615, 'Anub\'arak (2)');
(21601, 35449, 'Icehowl (3)');
(29240, 35352, 'Fjola Lightbane (3)');
(29267, 35349, 'Eydis Darkbane (3)');
(29268, 35616, 'Anub\'arak (3)');
(30362, 36853, 'Sindragosa');
(29524, 35360, 'Koralon the Flame Watcher (1)');
(31089, 38462, 'Toravon the Ice Watcher (1)');
(32179, 39746, 'General Zarithrian');
(31761, 39751, 'Baltharus the Warborn');
(31952, 39863, 'Halion');
(31761, 39899, 'Baltharus the Warborn');
(31952, 40142, 'Halion');
(2718, 40870, 'Ruby Dragon');
(32179, 39805, 'General Zarithrian (1)');
(31952, 39864, 'Halion (1)');
(31761, 39920, 'Baltharus the Warborn (1)');
(31761, 39922, 'Baltharus the Warborn (1)');
(31952, 39944, 'Halion (2)');
(31952, 39945, 'Halion (3)');
(31952, 40143, 'Halion (1)');
(31952, 40144, 'Halion (2)');
(31952, 40145, 'Halion (3)');
(21174, 250006, 'Snake boss add two');
(21174, 250005, 'Slythe\'s Guardian');
(28641, 55555, 'Fall event starter');
(28641, 55556, 'Fall event questgivers');
(15945, 261000, 'Soul of Kel\'Thuzad');
(8570, 261003, 'Soul of Onyxia');
(29268, 261005, 'Soul of Anub\'arak');
(28951, 261018, 'Brain of Yogg-Saron');
```

**Comment by Aridam** [comment]
I tried to recreate this:
https://www.youtube.com/watch?v=8WnaIKa6weY&feature=youtu.be&t=1m59s
and got this:
https://gfycat.com/similarunconsciouschrysomelid
As you can see as per RangeDisplay, the distance from the farther Dummy is ~8yd (max). I couldn't Cleave while trying to face the same way as the Druid in the original video, but once I narrowed the angle spread between the two, it worked and chained to the farther Dummy (which I can't hit personally hit from where I am). In this particular case, angle is definitely off.

Gonna sift through your list and check for video footage on Cleave usage on/off those NPCs for Hitbox/Combat Reach regards.

**Comment by Riztazz** [comment]
And our current code checks if target is within 180 degrees of caster and math checks out - so blizzard had to add a little bit of leeway to this as well, so it's not so unforgiving

**Comment by Aridam** [comment]
I've also found this:
https://web.archive.org/web/20070402035846/http://forums.worldofwarcraft.com/thread.html?topicId=86009546&pageNo=1&sid=1
> The following changes to cleave have been made for the 2.1.0 patch.
--Cleaves will only hit people in front of the creature.
**_--Cleaves will have a chain length equal to the combat reach of the creature._**

**Comment by Riztazz** [comment]
So that pretty much confirms what i was saying, that output of base value + combat reach has to be limited to some value. Still gotta figure out base value for cleave

**Comment by Aridam** [comment]
Check this:
https://youtu.be/E166ukqzA80?t=88
Watch the whole Transition Phase. Look at the Cleaves, done by Felguards and the Warrior himself.

Here too:
https://www.youtube.com/watch?v=E166ukqzA80&feature=youtu.be&t=200

Here as well:
https://youtu.be/E166ukqzA80?t=310

**Comment by Riztazz** [comment]
Ice tombs have combat reach of 7 yards.
Another theory could be that minimum radius is either 8 or 10 yards (if theory is correct then it has to be 10) and if combat reach exceeds that value then combat reach value is used instead

**Comment by Aridam** [comment]
> Ice tombs have combat reach of 7 yards.

Ah, welp, okay. So yeah, nothing new compared to previous data.

> Another theory could be that minimum radius is either 8 or 10 yards (if theory is correct then it has to be 10) ...

How would it be 10yd if your theory is correct? Out of interest.

>... and if combat reach exceeds that value then combat reach value is used instead.

The v2.1.0 change -- if taken _literally_ -- would mean that Cleaves would _equal the **Target's Combat Reach**_. In a sense that would mean Cleave would be able to chain on/from Targets with gigantic Combat Reaches?

**Comment by Riztazz** [comment]
Because those dummies in orgrimmar are 9 yards(and change) apart from each other.

Answering your question:
That could mean a lot of things actually, maybe blizzard used to use a static radius for cleaves, ie. search targets in 50 yards and cleave them. Cleave used to work differently back then

**Comment by Aridam** [comment]
Maybe this helps?
https://www.youtube.com/watch?v=e4ZkhrwwGdw&feature=youtu.be&t=398
Look at that ridiculous Seal of Command proc -- it hits the Target, chains to the left one and then _further_ chains to the _far-left_ one. SoComm strikes up to two Targets by default.

**Comment by Riztazz** [comment]
I'll check the ranges between dummies in a bit

**Comment by Riztazz** [comment]
A little bit over 15 yards in between the main target of cleave and furthest dummy (left one).
I have no idea what is the rule for that spell :c

**Comment by Aridam** [comment]
I believe that one is able to hit behind you (so, all around you, in a 360 cone), and tries to find a Target to chain to. I think it's reminiscent of one of those way-old 360-degree "Cleave-chain" Spells (like some Vanilla Raid Trash/Boss Mobs would use).

**Comment by Riztazz** [comment]
What you're describing if far bouncing cleaves and i actually figured out what controls what.
SOC is one of those far bouncing, so is ie. chain heal (cleave as well, spell effect wise).
Far bouncing cleave spells bounce off of targets. Non far bouncing, like warriors cleave or heart strike bounce off main target and within specified radius, which we do not know yet, but from what i'm seeing it's 10y + combat reach BUT can't exceed that value if combat reach alone is bigger than that by default

**Comment by Aridam** [comment]
>What you're describing if far bouncing cleaves and i actually figured out what controls what.
SOC is one of those far bouncing, so is ie. chain heal (is cleave as well, spell effect wise).
Far bouncing cleave spells bounce off of targets. ...

Good explanation. That's entirely correct.

>... Non far bouncing, like warriors cleave or heart strike bounce off main target and within specified radius, which we do not know yet, but from what i'm seeing it's 10y + combat reach BUT can't exceed that value if combat reach alone is bigger than that by default

I'm still going off your list trying to find something that accurately shows this so that it can be used as proper data to determine this, but so far, nothing exceptional on my end (also had no Internet for like two hours).

**Comment by Riztazz** [comment]
Give it a spin after realm update, it's still lacking some data, but should behave way better now

**Comment by styxel** [comment]
Feral swipe already feels better at least on dummies actually hitting targets To your sides better now testing on raids on thursday..
